ducktape/command_line/parse_args.py

```diff
@@ -58,6 +58,12 @@ def create_ducktape_parser():
                         help="Upper bound on number of tests run simultaneously.")
     parser.add_argument("--repeat", action="store", type=int, default=1,
                         help="Use this flag to repeat all discovered tests the given number of times.")
+    parser.add_argument("--subsets", action="store", type=int, default=1,
```

Comment:
The idea was to keep this simple enough that it can be maintained easily. I realize there is variation in the test run times, that's why I listed out the times [in this comment](https://github.com/apache/kafka/pull/2172#issuecomment-263045897) and pointed out that if you select every Nth test instead of blocks of N, you avoid grouping together paramatrizations of the same test method that are likely to take similar (potentially long) amounts of time. Given how the test run times currently look, I'm not convinced we can't get a good distribution using a simple scheme like this.

The idea was to make it more likely you'll get an even distribution without having to manually maintain groupings of tests that are highly specific to the setup you're running the tests on. Using explicit ranges like you mention has a serious drawback as well -- when you add new tests, it's very easy to accidentally forget to update this mapping and then those tests won't even be running.

I'm actually not that happy about adding this as a fixed set of parameters to ducktape itself rather than, e.g., making the test runner pluggable or something like that so we have the flexibility to change this moving forward. But my biggest concern is that we do something that doesn't require constant maintenance and updates to the breakdown of tests into groups. If that means saving info about test run times and allowing the test runner to take advantage of that when breaking tests into groups, that'd be significantly preferable to anything that requires manually maintaining lists of tests...